utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -1,457 +1,210 @@
-
-    def num_x(self):
-        """num_x Width of image in pixels
-
-        Returns:
-            int: Width of image in pixels
-        """
-        return self._xyzct['X']
-
-    def num_y(self):
-        """num_y Height of image in pixels
-
-        Returns:
-            int: Height of image in pixels
-        """
-        return self._xyzct['Y']
-
-    def num_z(self):
-        """num_z Depth of image in pixels
-
-        Returns:
-            int: Depth of image in pixels
-        """
-        return self._xyzct['Z']
-
-    def num_c(self):
-        """num_c Number of channels in the image
-
-        Returns:
-            int: Number of channels
-        """
-        return self._xyzct['C']
-
-    def num_t(self):
-        """num_x Number of timepoints in an image
-
-        Returns:
-            int: Number of timepoints
-        """
-        return self._xyzct['T']
-
-    def physical_size_x(self):
-        """num_x Size of pixels in x-dimension
-
-        Returns:
-            float: Number of units per pixel
-            str: Units (i.e. cm or mm)
-        """
-        return (self._metadata.image(0).Pixels.PhysicalSizeX, self._metadata.image(0).Pixels.PhysicalSizeXUnit)
-
-    def physical_size_y(self):
-        """num_y Size of pixels in y-dimension
-
-        Returns:
-            float: Number of units per pixel
-            str: Units (i.e. cm or mm)
-        """
-        return (self._metadata.image(0).Pixels.PhysicalSizeY, self._metadata.image(0).Pixels.PhysicalSizeYUnit)
-
-    def physical_size_z(self):
-        """num_z Size of pixels in z-dimension
-
-        Returns:
-            float: Number of units per pixel
-            str: Units (i.e. cm or mm)
+        Args:
+            file_path: Path to file to read
+            max_workers: Number of threads used to read and image. *Default is
+                half the number of detected cores.*
+            backend: Must be ``python`` or ``java``. *Default is python.*
         """
-        return (self._metadata.image(0).Pixels.PhysicalSizeZ, self._metadata.image(0).Pixels.PhysicalSizeZUnit)
-
-    def read_metadata(self, update=False):
-        """read_metadata Get the metadata for the image
 
-        This function calls the Bioformats metadata parser, which extracts metadata
-        from an image. This returns the python-bioformats OMEXML class, which is a
-        convenient handler for the complex xml metadata created by Bioformats.
+        # Initialize BioBase
+        super(BioReader, self).__init__(file_path,
+                                        max_workers=max_workers,
+                                        backend=backend)
         
-        Most basic metadata information have their own BioReader methods, such as
-        image dimensions(i.e. num_x(), num_y(), etc). However, in some cases it may
-        be necessary to access the underlying metadata class.
+        # Ensure backend is supported
```

Comment:
Can you describe when its appropriate to use the Python backend or Java backend? 
I don't understand the difference

Reply:
In the documentation it goes over the differences. Inside of WIPP, you will likely only ever use the Python backend. The Python backend saves all data in tiled tiff format with compression automatically. The java backend actually needs to be used with the javabridge like we used to have to do for every plugin. The advantage of using the Java backend is that it can open any file format that that Bioformats can open (150+ formats), while the Python backend can only open the OME tiled tiff format. Since we automatically convert all data to tiled tiff format when data is uploaded to WIPP, there is generally no need to use the Java backend. We do have some special plugins that need to do some special operations on original data before converted to tiled tiff conversion, so that's where the Java backend is useful. It also helps when validating the Python backend. Basically, our Python backend needs to read data exactly like the Java backend, and if it doesn't we need to update the code to make sure it does.